_source/_posts/2024-01-24-spring-boot-authorization.md

```diff
@@ -0,0 +1,569 @@
+- Auth0 Client ID: \<client-id\>
+- Auth0 Callback URL: https://dashboard.whatabyte.app/home
+- Auth0 API Audience: https://menu-api.okta.com
+
+Click **Save**.
+
+For the API server to accept requests from the dashboard, you must tweak the CORS configuration, enabling the dashboard URL as an allowed origin for requests.
+
+In the API project, add a `SecurityConfig` class at the root package:
+
+```java
+// src/main/java/com/example/menu/SecurityConfig.java
+package com.example.menu;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.web.SecurityFilterChain;
+
+import static org.springframework.security.config.Customizer.withDefaults;
+
+@Configuration
+public class SecurityConfig {
+
+    @Bean
+    public SecurityFilterChain configure(HttpSecurity http) throws Exception {
+        return http
+                .oauth2ResourceServer(oauth2ResourceServer -> oauth2ResourceServer.jwt(withDefaults()))
+                .build();
+    }
+
+}
+```
+
+Modify `ItemController` by adding the `@CrossOrigin` annotation:
+
+```java
+// src/main/java/com/example/menu/web/ItemController.java
+...
+
+@RestController
+@RequestMapping("/api/menu/items")
+@CrossOrigin(origins = "https://dashboard.whatabyte.app")
+public class ItemController {
+
+...
+```
+
+Restart the server API. Before the sign in, you can create some test users with the Auth0 CLI. This step is optional, you can sign up later from the [Auth0 Universal Login](https://auth0.com/docs/authenticate/login/auth0-universal-login) form, or choose Google social login.
+
+```shell
+auth0 users create
+```
+
+Complete the required information, you will see an output like the lines below:
+
+```text
+Connection Name: Username-Password-Authentication
+ Name: Thomas
+ Email: thomas@email.com
+ Password: ********
+
+=== dev-avup2laz.us.auth0.com user created
+
+  ID          auth0|643ec0e1e671c7c9c5916ed6
+  EMAIL       thomas@email.com
+  CONNECTION  Username-Password-Authentication
+```
+Save the ID for later.
+
+In the dashboard, click the **Sign in** button. The client will redirect you to the [Auth0 Universal Login](https://auth0.com/docs/authenticate/login/auth0-universal-login) page. Sign in with the user you just created. In the left menu, choose **Menu**, and the menu items will display.
+
+{% img blog/spring-boot-authorization/whatabyte-ui.png alt:"WHATABYTE Dashboard" width:"800" %}{: .center-image }
+
+On the top right corner, click **Add Item**, and a pre-populated form will display. Click on **Save** and verify the user request is authorized in the server.
+
+## Role-Based Access Control (RBAC)
+
+If you disable the authentication features in the WHATABYTE client, it will allow you to attempt CRUD operations from the UI, but your API server will reject item modifications as an unauthenticated user.
```

Comment:
I think this is the dashboard. can we use either dashboard or WHATABYTE client consistently to avoid confusion

Reply:
Updated